rest/route/patch.go

```diff
@@ -484,3 +487,124 @@ func (p *mergePatchHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 	return gimlet.NewJSONResponse(apiPatch)
 }
+
+type patchTasks struct {
+	Description string    `json:"description"`
+	Variants    []variant `json:"variants"`
+}
+
+type variant struct {
+	Id    string   `json:"id"`
+	Tasks []string `json:"tasks"`
+}
+
+type schedulePatchHandler struct {
+	variantTasks patchTasks
+
+	patchId string
+	patch   patch.Patch
+	sc      data.Connector
+}
+
+func makeSchedulePatchHandler(sc data.Connector) gimlet.RouteHandler {
+	return &schedulePatchHandler{
+		sc: sc,
+	}
+}
+
+func (p *schedulePatchHandler) Factory() gimlet.RouteHandler {
+	return &schedulePatchHandler{
+		sc: p.sc,
+	}
+}
+
+func (p *schedulePatchHandler) Parse(ctx context.Context, r *http.Request) error {
+	p.patchId = gimlet.GetVars(r)["patch_id"]
+	if p.patchId == "" {
+		return errors.New("must specify a patch ID")
+	}
+	var err error
+	apiPatch, err := p.sc.FindPatchById(p.patchId)
+	if err != nil {
+		return err
+	}
+	if apiPatch == nil {
+		return errors.New("patch not found")
+	}
+	dbPatch, err := apiPatch.ToService()
+	if err != nil {
+		return errors.Wrap(err, "unable to parse patch")
+	}
+	p.patch = dbPatch.(patch.Patch)
+	body := util.NewRequestReader(r)
+	defer body.Close()
+	patchTasks := patchTasks{}
+	if err = utility.ReadJSON(body, &patchTasks); err != nil {
+		return errors.Wrap(err, "Argument read error")
+	}
+	if len(patchTasks.Variants) == 0 {
+		return errors.New("no variants specified")
+	}
+	p.variantTasks = patchTasks
+	return nil
+}
+
+func (p *schedulePatchHandler) Run(ctx context.Context) gimlet.Responder {
+	settings, err := evergreen.GetConfig()
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "unable to get config"))
+	}
+	token, err := settings.GetGithubOauthToken()
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "unable to get token"))
+	}
+	dbVersion, _ := p.sc.FindVersionById(p.patchId)
+	var project *dbModel.Project
+	grip.Warning(p.patch)
+	if dbVersion == nil {
+		project, _, err = dbModel.GetPatchedProject(ctx, &p.patch, token)
+		if err != nil {
+			return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "unable to find project from patch"))
+		}
+	} else {
+		project, err = dbModel.FindProjectFromVersionID(dbVersion.Id)
+		if err != nil {
+			return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "unable to find project from version"))
+		}
+	}
+	variantTasks := graphql.PatchVariantsTasksRequest{}
```

Comment:
Is it weird to be using graphql to be doing completely backend things? Are we worried that having this package depend on graphql could lead to a strange dependency in the future? I think maybe PatchVariantsTasksRequest doesn't belong in graphql, and graphql should import it from a model package or something.

Reply:
a bit, but I don't think it's too awkward. Right now it's logic that was previously in the service package, but is now shared so it's not any worse than what was before. Arguably this specific function should be in model, but for the most part I think it's fine